### PR TITLE
bugfix: settings rollback after modifying settings of selected server and then clicking "apply" button.

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
What does this implement/fix? Explain your changes.
---------------------------------------------------
* upgrade intellij gradle plugin to v1.10.0
* use `Class.isAssignableFrom` instead of `equals` to get right DBMS for newly created sql db servers, because newly created server instances are `XxxDraft`s..
* use `jdbcUrl` as single source of truth for DbServerParamEditor and simplify code.

Does this close any currently open issues?
------------------------------------------
https://github.com/microsoft/azure-tools-for-java/commit/2a4aadc32205248f5a8f717ac83c3a857abf7488 [AB#2008980](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/2008980): [Test]The database tool window has no server option and the sql does not match the actual

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-database/src/main/java/com/microsoft/azure/toolkit/intellij/database/dbtools/OpenWithDatabaseToolsAction.java [hunk 1]
```diff
@@ -32,13 +32,13 @@ public static void openDataSourceManagerDialog(IDatabaseServer<?> server, Projec
     }
 
     private static Dbms getDbms(IDatabaseServer<?> server) {
-        if (MicrosoftSqlServer.class.equals(server.getClass())) {
+        if (MicrosoftSqlServer.class.isAssignableFrom(server.getClass())) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-database/src/main/java/com/microsoft/azure/toolkit/intellij/database/dbtools/OpenWithDatabaseToolsAction.java]: use `instanceof`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-database/src/main/resources/META-INF/azure-intellij-plugin-database-dbtools.xml [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <idea-plugin>
     <extensions defaultExtensionNs="com.intellij.database">
-        <connectionInterceptor implementation="com.microsoft.azure.toolkit.intellij.database.dbtools.DatabaseServerConnectionInterceptor"/>
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-database/src/main/resources/META-INF/azure-intellij-plugin-database-dbtools.xml]: add back this to record data source connection.

**Review by ivywei0125 — APPROVED** [review]
LGTM

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**